run_distillation.py

```
@@ -208,472 +239,615 @@ def parse_args_to_config() -> TrainingConfig:
+                    torch.cuda.set_device(prev_device)
+            free_gb = free_bytes / (1024**3)
+            gpu_free_mem[idx] = (free_gb, total_bytes / (1024**3))
+            if free_gb > min_free_gb and free_gb > best_free_gb:
+                best_gpu = idx
+                best_free_gb = free_gb
+
+        if best_gpu is None:
+            details = ", ".join(f"{i}:{free:.1f}GB" for i, (free, _) in gpu_free_mem.items())
+            raise RuntimeError(
+                f"No GPUs with >{min_free_gb:.0f}GB free memory available. Free memory (GB): [{details}]"
+            )
```

Ruff (0.14.1)

Avoid specifying long messages outside the exception class

(TRY003)